lib/cyclonedx/cocoapods/component.rb

```diff
@@ -53,13 +53,24 @@ def initialize(name:, version:, type:, group: nil, build_system: nil, vcs: nil)
         @name = name
         @version = version
         @type = type
+        # Add these instance variables before the bomref assignment
+        @bomref = build_purl(type, name, group, version)
         @build_system = build_system
         @vcs = vcs
-        @bomref = "#{name}@#{version}"
+      end
 
-        return if group.nil?
+      private
 
-        @bomref = "#{group}/#{@bomref}"
+      def build_purl(type, name, group, version)
+        package_type = type == 'library' ? 'cocoapods' : 'generic'
+        if group.nil?
+          purl_name, subpath = parse_name(name)
+        else
+          # this seems wrong as cocoapods does not use groups?
```

Comment:
This is the part I am not comfortable with for a `library` style `purl` here.  If we're analyzing an application, then `pkg:generic/<group>/<name>@<version>#<subpath>` is correct I think.  The caller enters the group, name, and version as part of the CLI to invoke this, and then we just say it's a `pkg:generic` purl that may or may not be in anybody else's databases.

However, if the thing we are analyzing is itself a `library` that is available through `CocoaPods` and if the `bom-ref` is a `purl`, I would expect it to be the correct purl of the pod being analyzed and **not** up to the CLI parameters being passed.  The information to build the `purl` would be in a `*.podspec` file where the name of the file is the name of the pod.  We don't currently have code here to read a `podspec` file, but it is part of the ruby library that we are importing from the CocoaPods project.